llms/hf_llm/models.py

```diff
@@ -73,8 +82,17 @@ def __init__(self, args: ModelArgs):
         self.k_proj = nn.Linear(dim, n_kv_heads * head_dim, bias=False)
         self.v_proj = nn.Linear(dim, n_kv_heads * head_dim, bias=False)
         self.o_proj = nn.Linear(n_heads * head_dim, dim, bias=False)
+        rope_scale = (
+            1 / args.rope_scaling.get("factor", 1.0)
+            if args.rope_scaling is not None
+            and args.rope_scaling.get("type") == "linear"
```

Comment:
Also I would just do the following since you have to have a factor if the type is linear, and you have to have a type.

```suggestion
            1 / args.rope_scaling["factor"]
            if args.rope_scaling is not None
            and args.rope_scaling["type"] == "linear"
```

Reply:
Yeah, it's just a habit of my defensive coding style. I'm happy to change the preferred style.

Reply:
Yea exactly.. and I was also thinking if we get to this point and those values don't exist.. crashing is probably a good thing.